f5/bigip/tm/ltm/test/test_auth.py

```diff
@@ -0,0 +1,179 @@
+# limitations under the License.
+#
+
+
+from f5.bigip import ManagementRoot
+from f5.bigip.resource import MissingRequiredCreationParameter
+from f5.bigip.resource import UnsupportedMethod
+from f5.bigip.tm.ltm.auth import Crldp_Server
+from f5.bigip.tm.ltm.auth import Kerberos_Delegation
+from f5.bigip.tm.ltm.auth import Ldap
+from f5.bigip.tm.ltm.auth import Ocsp_Responder
+from f5.bigip.tm.ltm.auth import Profile
+from f5.bigip.tm.ltm.auth import Radius
+from f5.bigip.tm.ltm.auth import Radius_Server
+from f5.bigip.tm.ltm.auth import Ssl_Cc_Ldap
+from f5.bigip.tm.ltm.auth import Ssl_Crldp
+from f5.bigip.tm.ltm.auth import Ssl_Ocsp
+from f5.bigip.tm.ltm.auth import Tacacs
+import mock
+import pytest
+
+
+class HelperTest(object):
+    def __init__(self, collection_name):
+        self.partition = 'Common'
+        self.lowered = collection_name.lower()
+        self.test_name = 'fake_' + self.urielementname()
+        self.authkinds = {
+            'crldp_server': 'tm:ltm:auth:crldp-server:crldp-serverstate',
+            'kerberos_delegation':
+                'tm:ltm:auth:kerberos-delegation:kerberos-delegationstate',
+            'ldap': 'tm:ltm:auth:ldap:ldapstate',
+            'ocsp_responder': 'tm:ltm:auth:ocsp-responder:ocsp-responderstate',
+            'profile': 'tm:ltm:auth:profile:profilestate',
+            'radius': 'tm:ltm:auth:radius:radiusstate',
+            'radius_server': 'tm:ltm:auth:radius-server:radius-serverstate',
+            'ssl_cc_ldap': 'tm:ltm:auth:ssl-cc-ldap:ssl-cc-ldapstate',
+            'ssl_crldp': 'tm:ltm:auth:ssl-crldp:ssl-crldpstate',
+            'ssl_ocsp': 'tm:ltm:auth:ssl-ocsp:ssl-ocspstate',
+            'tacacs': 'tm:ltm:auth:tacacs:tacacsstate'
+            }
+
+    def urielementname(self):
+        if self.lowered[-2:] == '_s':
+            endind = 2
+        else:
+            endind = 1
+        return self.lowered[:-endind]
+
+    def set_resources(self, fakeicontrolsession_v12):
+        mr = ManagementRoot('192.168.1.1', 'admin', 'admin')
+        resourcecollection =\
+            getattr(getattr(getattr(mr.tm, 'ltm'), 'auth'),
+                    self.lowered)
+        resource1 = getattr(resourcecollection, self.urielementname())
+        resource2 = getattr(resourcecollection, self.urielementname())
+        return resource1, resource2
+
+    def set_collections(self, fakeicontrolsession_v12):
+        mr = ManagementRoot('192.168.1.1', 'admin', 'admin')
+        resourcecollection =\
+            getattr(getattr(getattr(mr.tm, 'ltm'), 'auth'),
+                    self.lowered)
+        return resourcecollection
+
+    def test_collections(self, fakeicontrolsession_v12, klass):
+        rc = self.set_collections(fakeicontrolsession_v12)
+        test_meta = rc._meta_data['attribute_registry']
+        test_meta2 = rc._meta_data['allowed_lazy_attributes']
+        kind = self.authkinds[self.urielementname()]
+        assert kind in test_meta.keys()
```

Comment:
change this to list(iterkeys(test_meta)) and include "from six import iterkeys" at the top of the file for python 3